File: performance/memory_allocation.lua
local function create_array(size)
  local arr = {}
  for i = 1, size do
    arr[i] = i - 1
  end
  return arr
end

math.randomseed(42)
local total_length = 0

for i = 0, 999 do
  local size = math.random(10, 50)
  local arr = create_array(size)

  local bounded_len = total_length
  local bounded_arr_len = #arr
  total_length = bounded_len + bounded_arr_len

  local temp1 = {1, 2, 3, 4, 5}
  local temp2 = {}
  for _, value in ipairs(temp1) do
    temp2[#temp2 + 1] = value
  end
  for j = 1, math.min(5, #arr) do
    temp2[#temp2 + 1] = arr[j]
  end

  local bounded_len2 = total_length
  local bounded_temp_len = #temp2
  total_length = bounded_len2 + bounded_temp_len
end

print(total_length)

File: performance/memory_allocation.py
import random

def create_array(size):
    arr = [0] * size
    for i in range(size):
        arr[i] = i
    return arr

def main():
    random.seed(42)
    total_length = 0

    # Memory allocation and array creation benchmark
    for i in range(1000):
        size = random.randint(10, 50)
        arr = create_array(size)

        bounded_len = total_length
        bounded_arr_len = len(arr)
        total_length = (bounded_len + bounded_arr_len)

        # Create some temporary arrays
        temp1 = [1, 2, 3, 4, 5]
        temp2 = temp1 + arr[0:5]
        bounded_len2 = total_length
        bounded_temp_len = len(temp2)
        total_length = (bounded_len2 + bounded_temp_len)

    print(total_length)

main()